//syntax = "proto3";
//
//package com.cognix;
//option go_package = "backend/core/proto;proto";
//
//message EmbedRequest {
//    repeated string contents = 1; // Changed to repeated for batching
//    string model = 2;
//}
//
//message EmbedResponseItem {
//    string content = 1;
//    repeated float vector = 2;
//}
//
//message EmbedResponse {
//    repeated EmbedResponseItem embeddings = 1; // List of embedding items
//}
//
//service EmbedService {
//    rpc GetEmbedding (EmbedRequest) returns (EmbedResponse) {}
//}


syntax = "proto3";

package com.cognix;
option go_package = "backend/core/proto;proto";

message EmbedRequest {
    string content = 1;
    string model = 2;
}

message EmbedResponse {
    repeated float vector = 1;
}

service EmbedService {
    rpc GetEmbeding (EmbedRequest) returns (EmbedResponse) {}
}
